### PR TITLE
Vanagon should be able to appropriately add in packaging scripts

### PR DESCRIPTION
**Pull request by melissa**

### CODE REVIEW

**Comment by haus** [comment]
CPR-235 looks unrelated to rpm from reading the ticket.

**Comment by melissa** [comment]
This is only the rpm half. I'll either update this PR with the deb half, or open a new one if this is merged.

**Comment by melissa** [comment]
centos 7 razor scripts
```
# rpm -qp --scripts razor-0.0.1.13.g9a3c714-1.el7.x86_64.rpm
preinstall scriptlet (using /bin/sh):
getent group razor >/dev/null || groupadd -r razor
getent passwd razor >/dev/null ||       useradd -r -g razor -d /var/lib/razor -s /bin/bash       -c "Razor service account" razor
exit 0
postinstall scriptlet (using /bin/sh):
  # switch based on systemd vs systemv vs smf vs aix
  #

if [ $1 -eq 1 ] ; then
        # Initial installation
        /usr/bin/systemctl preset razor-server.service >/dev/null 2>&1 || :
fi
preuninstall scriptlet (using /bin/sh):
# Undeploy our application from TorqueBox.
source /etc/razor/razor-torquebox.sh
torquebox undeploy /opt/razor || :



if [ $1 -eq 0 ] ; then
        # Package removal, not upgrade
        /usr/bin/systemctl --no-reload disable razor-server.service > /dev/null 2>&1 || :
        /usr/bin/systemctl stop razor-server.service > /dev/null 2>&1 || :
fi
postuninstall scriptlet (using /bin/sh):

  # switch based on systemd vs systemv vs smf vs aix
  #

/usr/bin/systemctl daemon-reload >/dev/null 2>&1 || :
if [ $1 -ge 1 ] ; then
        # Package upgrade, not uninstall
        /usr/bin/systemctl try-restart razor-server.service >/dev/null 2>&1 || :
fi
posttrans scriptlet (using /bin/sh):
    # Cause our application to be deployed to TorqueBox, but only the first time
# we are installed.  Upgrades do not automatically trigger this, because a
# database migration may be required, and we want to avoid reloading code in a
# way that will break compared to that "admin intervention required"
# planned downtime.
if [ x"$1" = x"1" ]; then
    source /etc/razor/razor-torquebox.sh
    torquebox deploy /opt/razor --env=production
fi
exit 0
```

**Comment by melissa** [comment]
centos 6 razor scripts
```
# rpm -qp --scripts razor-0.0.1.13.g9a3c714-1.el6.x86_64.rpm
preinstall scriptlet (using /bin/sh):
getent group razor >/dev/null || groupadd -r razor
getent passwd razor >/dev/null ||       useradd -r -g razor -d /var/lib/razor -s /bin/bash       -c "Razor service account" razor
exit 0
postinstall scriptlet (using /bin/sh):
  # switch based on systemd vs systemv vs smf vs aix
  #
    chkconfig --add razor-server >/dev/null 2>&1 || :
preuninstall scriptlet (using /bin/sh):
# Undeploy our application from TorqueBox.
source /etc/razor/razor-torquebox.sh
torquebox undeploy /opt/razor || :


    if [ $1 = 0 ]; then
      /sbin/service razor-server stop >/dev/null 2>&1 || :
      chkconfig --del razor-server || :
    fi
postuninstall scriptlet (using /bin/sh):

  # switch based on systemd vs systemv vs smf vs aix
  #
    if [ $1 -ge 1 ]; then
      /sbin/service razor-server condrestart || :
    fi
posttrans scriptlet (using /bin/sh):
    # Cause our application to be deployed to TorqueBox, but only the first time
# we are installed.  Upgrades do not automatically trigger this, because a
# database migration may be required, and we want to avoid reloading code in a
# way that will break compared to that "admin intervention required"
# planned downtime.
if [ x"$1" = x"1" ]; then
    source /etc/razor/razor-torquebox.sh
    torquebox deploy /opt/razor --env=production
fi
exit 0
```

**Comment by melissa** [comment]
Ah, I see what you mean, @haus. Commit message updated.

**Comment by shrug** [comment]
@melissa This needs a rebase?

File: templates/deb/control.erb [hunk 1]
```diff
@@ -15,7 +15,7 @@ Conflicts: <%= get_replaces.map { |replace| "#{replace.replacement} #{replace.ve
 Depends: <%= get_requires.join(", ") %>
 Provides: <%= get_provides.map { |prov| prov.provide }.join(", ") %>
 Description: <%= @description.lines.first.chomp %>
- <%= @description.lines[1..-1].join("\n\s") -%>
+ <%= @description.lines[1..-1].join("\s") -%>
```
Comment (haus) [inline templates/deb/control.erb]: This change should get pulled out into a separate PR if possible.

File: lib/vanagon/component/dsl.rb [hunk 2]
```diff
@@ -289,18 +289,44 @@ def environment(env)
...
-      # Adds actions to run at the beginning of a package install
+      # Adds action to run during the preinstall phase of packaging
       #
-      # @param action [String] Bourne shell compatible scriptlets to execute
-      def add_preinstall_action(action)
-        @component.preinstall_actions << action
+      # @param pkg_state [Array] the state in which the scripts should execute. Can be
+      #   one or multiple of 'install' and 'upgrade'.
+      # @param block [Array] the Bourne shell compatible scriptlet(s) to execute
+      def add_preinstall_action(pkg_state, block)
```
Comment (stahnma) [inline lib/vanagon/component/dsl.rb]: Why have the add_ in the method name? Couldn't it just be preinstall_action? I don't think we have a corresponding remove_preinstall_action.

**Comment by melissa** [comment]
closed in favor of https://github.com/puppetlabs/vanagon/pull/207

File: lib/vanagon/component/dsl.rb (hunk 2 above)
Comment (melissa) [inline lib/vanagon/component/dsl.rb]: Is there a case where we would want to remove a packaging action? Add to me makes the most sense (i.e., https://github.com/melissa/razor-vanagon/blob/master/configs/components/razor-server.rb#L86-L90). It follows established convention, like `install_configfile`, since it can be called multiple times
